python/Ganga/Core/GangaRepository/GangaRepositoryXML.py

```diff
@@ -225,7 +225,7 @@ def shutdown(self):
         from Ganga.Utility.logging import getLogger
         logger = getLogger()
         logger.debug("Shutting Down GangaRepositoryLocal: %s" % self.registry.name)
-        for k in self._fully_loaded.keys():
+        for k in [_k for _k in self._fully_loaded.keys() if _k not in self.incomplete_objects]:
```

Comment:
Yes, but we should never be risking user data on disk ever. I'm still paranoid that something will come along and ruin my jobs as it is still doing.